control_panel_api/views.py

```diff
@@ -108,14 +177,27 @@ class S3BucketViewSet(viewsets.ModelViewSet):
     filter_backends = (S3BucketFilter,)
     permission_classes = (S3BucketPermissions,)
 
+    @transaction.atomic
     def perform_create(self, serializer):
         instance = serializer.save(created_by=self.request.user)
-        instance.aws_create()
+
+        try:
+            instance.aws_create()
+        except ClientError as e:
```

Comment:
As [said in other comment](https://github.com/ministryofjustice/analytics-platform-control-panel/pull/67/files#r152300693), we may want to be specific when catching this exception to not roll back errors caused by us adding an S3 bucket already existent in the API.